compute/computer.go

```diff
@@ -177,17 +178,40 @@ func (c *Computer) ClearCacheForProject(projectID uuid.UUID) {
 	c.cache.Delete(projectID)
 }
 
-func newValueHandler(states AccountStates) func(owner flow.Address, key string, value cadence.Value) error {
-	return func(owner flow.Address, key string, value cadence.Value) error {
-		// TODO: Remove address conversion
-		address := model.NewAddressFromBytes(owner.Bytes())
+func extractStateChangesFromDelta(d delta.Delta) AccountStates {
+	states := make(AccountStates)
+
+	ids, values := d.RegisterUpdates()
+
+	for i, id := range ids {
+		addressBytes := []byte(id.Owner)
+		if len(addressBytes) != flow.AddressLength {
+			continue
+		}
+		commonAddress := common.BytesToAddress(addressBytes)
 
-		if _, ok := states[address]; !ok {
-			states[address] = make(map[string]cadence.Value)
+		storedData, version := interpreter.StripMagic(values[i])
+		storedValue, err := interpreter.DecodeValue(storedData, &commonAddress, []string{id.Key}, version, nil)
+		if err != nil {
+			// skip this value, as it cannot be decoded
+			continue
 		}
 
-		states[address][key] = value
+		modelAddress := model.NewAddressFromBytes(addressBytes)
 
-		return nil
+		if _, ok := states[modelAddress]; !ok {
+			states[modelAddress] = make(map[string]cadence.Value)
+		}
+
+		inter, err := interpreter.NewInterpreter(nil, common.StringLocation("extractStateChangesFromDelta"))
+
+		if err != nil {
+			panic(err)
+		}
+
+		// How to convert interpreter.Value to cadence.Value
```

Comment:
@turbolent I tried simply removing *high level storage* and running the playground by having the `states` always empty. That gave me the following error on the playground whenever I tried to deploy a contract:

```
Error: GraphQL error: failed to get or create user: failed to load user from session: entity not found
``` 

So I tried using the delta to actually populate the `states`  after the transaction is executed. That would work, but I don't know how I could convert a `interpreter.Value` to a `cadence.Value`. It seems like I would need an interpreter instance, but I do not know how to create one here. It should be possible since I have access to the storage.

Reply:
Do you mean that this error is a client-side / frontend error? If so, then we should look into why/where it errors and fix that

Reply:
As for the other question, how we could extract the high-level values from the delta:
Getting the data, stripping the magic, and decoding the encoded value is a good start.

However, this is replicating much of how Cadence internally stores the data, maybe we can export the logic that does this in Cadence, `runtimeStorage.readValue` so it could be used here?

To export a value, `runtime.ExportValue` can be used. Like you noted, it requires an interpreter instance. It is needed, because some dictionaries' values are stored under separate storage keys. Exporting will load them from storage, so the passed interpreter must have a StorageReadHandler set on it. In the runtime this is `runtimeStorage.readValue`.

It might be a good idea to add another function to Cadence's runtime, which allows reading stored values, just like its new `ReadValue` function, but you can pass a low-level storage key (here, `id.Key`), instead of a high-level path.

Reply:
Thanks for the tip. That is what I needed to get it working.

What do you think of the current solution? I can add some error logging and better error handling if you like this solution.

There is a PR up for simply removing it https://github.com/onflow/flow-playground-api/pull/10 if this is too hacky.

Reply:
The current solution looks good! I think it's a good idea to use `ReadValue` here. It's a bit unfortunate the whole `runtime.Interface` has to be implemented, but I think it's fine 👍